modules/kafka/src/main/scala/higherkindness/mu/kafka/ConsumerStream.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017-2020 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.mu.kafka
+
+import cats.effect.{ConcurrentEffect, ContextShift, Sync, Timer}
+import fs2.Stream
+import fs2.kafka.{ConsumerSettings, KafkaConsumer}
+import higherkindness.mu.format.Decoder
+import io.chrisdavenport.log4cats.Logger
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+
+object ConsumerStream {
+
+  def apply[F[_]: Sync, A](topic: String, settings: ConsumerSettings[F, String, Array[Byte]])(
+      implicit contextShift: ContextShift[F],
+      concurrentEffect: ConcurrentEffect[F],
+      timer: Timer[F],
+      decoder: Decoder[A],
+      sync: Sync[F]
+  ): Stream[F, A] =
+    for {
+      implicit0(logger: Logger[F]) <- fs2.Stream.eval(Slf4jLogger.create[F])
+      s                            <- apply(fs2.kafka.consumerStream(settings))(topic)
+    } yield s
+
+  private[kafka] def apply[F[_], A](
+      kafkaConsumerStream: Stream[F, KafkaConsumer[F, String, Array[Byte]]]
+  )(topic: String)(
+      implicit contextShift: ContextShift[F],
+      concurrentEffect: ConcurrentEffect[F],
+      timer: Timer[F],
+      decoder: Decoder[A],
+      logger: Logger[F]
+  ): Stream[F, A] =
+    kafkaConsumerStream
+      .evalTap(_.subscribeTo(topic))
+      .flatMap(
+        _.stream
+          .flatMap { message =>
+            val a = decoder.decode(message.record.value)
+            for {
+              _ <- fs2.Stream.eval(logger.info(a.toString))
+            } yield a
```

Comment:
We can use `evalTap` in the logger print
```scala
kafkaConsumerStream
  .evalTap(_.subscribeTo(topic))
  .flatMap(_.stream.flatMap(msg => decoder.decode(msg.record.value)))
  .evalTap(a => logger.info(a.toString))
```

Reply:
On the other hand, I don't know if this (print the message) is something we want to do at library level or at library client level

Reply:
Yes, I know what you meant. In general it would be better if the library does less but it would be such a useful thing to have for debugging purpose to ensure that everything is working at the library level. We could put it at DEBUG or even at TRACE level.

Reply:
Setting the log level to `DEBUG` looks good to me. Thanks!

Reply:
I would not log the message contents at the library level, even as DEBUG. The message contents might be huge, and may contain sensitive data that shouldn't be logged. Best to let the user decide what they want to log.